tgui/packages/tgui/interfaces/LightController.tsx

```diff
@@ -0,0 +1,319 @@
+  category_ids: CategoryList;
+};
+
+export const LightController = (props, context) => {
+  const { act, data } = useBackend<Data>(context);
+  const { light_info, templates, default_id, default_category, category_ids } = data;
+  const [currentTemplate, setCurrentTemplate] = useLocalState<string>(
+    context,
+    'currentTemplate',
+    default_id
+  );
+  const [currentCategory, setCurrentCategory] = useLocalState<string>(
+    context,
+    'currentCategory',
+    default_category
+  );
+
+
+  return (
+    <Window
+      title={light_info.name + ": Lighting"}
+      width={600}
+      height={400}>
+      <Window.Content scrollable>
+        <Stack fill>
+          <Stack.Item>
+            <Section fitted fill scrollable width="170px">
+              <Tabs fluid centered>
+                {Object.keys(category_ids).map(
+                  (category, index) => (
+                    <Tabs.Tab
+                      key={category}
+                      selected={currentCategory === category}
+                      onClick={() => setCurrentCategory(category)}>
+                      <Box fontSize="14px" bold textColor={"#eee"}>
+                        {category}
+                      </Box>
+                    </Tabs.Tab>
+                  ))}
+              </Tabs>
+              <Tabs vertical>
+                {category_ids[currentCategory].map(
+                  (id) => (
+                    <Tabs.Tab
+                      key={id}
+                      selected={currentTemplate === id}
+                      onClick={() => setCurrentTemplate(id)}>
+                      <Box fontSize="14px" textColor={"#cee"}>
+                        {templates[id].light_info.name}
+                      </Box>
+                      <Box ml={0.1} className={classes(['lights32x32', "light_fantastic_" + id])} />
+                    </Tabs.Tab>
+                  ))}
+              </Tabs>
+            </Section>
+          </Stack.Item>
+          <Stack.Item grow>
+            <LightControl info={light_info} />
+            <LightInfo light={templates[currentTemplate]} />
+          </Stack.Item>
+        </Stack>
+      </Window.Content>
+    </Window>
+  );
+};
+
+type LightControlProps = {
+  info : LightDetails;
+};
+
+const LightControl = (props : LightControlProps, context) => {
+  const { act, data } = useBackend<Data>(context);
+  const { on } = data;
+  const { info } = props;
+  return (
+    <Section>
+      <Stack vertical justify="space-between" fill>
+        <Stack.Item>
+          <Stack justify="space-between">
+            <Stack.Item>
+              <Box fontSize="16px" mt={0.5}>
+                {info.name}
+              </Box>
+            </Stack.Item>
+            <Stack.Item>
+              <Button
+                fontSize="16px"
+                icon="brush"
+                textColor={info.color}
+                onClick={() => act("change_color")}
+                content={info.color}
```

Comment:
w/ buttons, please put the content as a child rather than as content
(if this causes textcolor to not work, nvm)

Reply:
No I just thought this was preferred is all